src/options.ts

```diff
@@ -71,6 +71,8 @@ export function resolveOptions(userOptions: UserOptions, viteRoot?: string): Res
 
   const extensions = userOptions.extensions || resolver.resolveExtensions()
 
+  extensions.sort((a, b) => b.length - a.length)
```

Comment:
Suppose the extensions option is set to `["tsx", "page.tsx"]`.
Because `find` function searches the array from the head, if there is a file `foo.page.tsx`, the first `tsx` matches and its path will be `foo.page`. But users generally expect to match with `page.tsx` and its path will be `foo`.

But yes, this is maybe a bit confusing feature. Instead, we should add documentation about the extension priority so that the users are aware of it.

Reply:
> Sort by string length to find the longest extension later

So you mean to find the longest extension first.

> Because find function searches the array from the head, if there is a file foo.page.tsx, the first tsx matches and its path will be foo.page. But users generally expect to match with page.tsx and its path will be foo.

Got it. but still I think this is a little bit confused, so let's add some doucmentaiton about extension priority and remove this extension sort.